<!-- -->
<launch>
  <arg name="maxVelocity"         default="1.2" />
  <arg name="minVelocity"         default="0.0" />
  <arg name="velocityProfileFactor"     default="1.0" />
  <arg name="maxPlanningDistance"     default="1000000.0" />
  <arg name="maxLocalPlanDistance"     default="40.0" />
  <arg name="samplingTipMargin"       default="1.5" />
  <arg name="samplingOutMargin"       default="4.5" />
  <arg name="samplingSpeedFactor"     default="0.25" />
  <arg name="pathDensity"         default="0.25" />
  <arg name="rollOutDensity"         default="0.25" />
  <arg name="rollOutsNumber"         default="6" />
  <arg name="horizonDistance"       default="10" />
  
  <arg name="minFollowingDistance"     default="10" />
  <arg name="maxFollowingDistance"     default="9" />
  <arg name="minDistanceToAvoid"       default="6" />
  <arg name="maxDistanceToAvoid"       default="3" />
  <arg name="speedProfileFactor"      default="1.0" />
  
  <arg name="enableSwerving"         default="true" />
  <arg name="enableFollowing"       default="true" />
  <arg name="enableHeadingSmoothing"     default="false" />
  <arg name="enableTrafficLightBehavior"   default="true" />
  <arg name="enableLaneChange"       default="false" />
  
  <arg name="width"             default="0.7" />
  <arg name="length"             default="1.2" />
  <arg name="wheelBaseLength"       default="1.0" />
  <arg name="turningRadius"        default="2.5" />
  <arg name="maxSteerAngle"         default="1.0" />
  
  <arg name="enableObjectTracking"     default="true" /> <!-- use internal kalman filter for track obstacles " -->
  <arg name="enableOutsideControl"     default="false" /> <!-- use internal kalman filter for track obstacles " -->
  <arg name="enableOdometryStatus"    default="true" /> <!-- read velocities from Odometry from robot "" -->
    
  <arg name="mapSource"           default="2" /> <!-- Autoware=0, Vector Map Folder=1, kml=2 -->
  <arg name="mapFileName"         default="/home/user/SimuLogs/KmlMaps/TsukubaMapWithStopLines.kml" />
  
  <node pkg="dp_planner" type="dp_planner" name="dp_planner" output="screen">
    <param name="maxVelocity"           value="$(arg maxVelocity)" />
      <param name="minVelocity"           value="$(arg minVelocity)" />
      
      <param name="velocityProfileFactor"     value="$(arg velocityProfileFactor)" />
    <param name="maxPlanningDistance"       value="$(arg maxPlanningDistance)" />
    <param name="maxLocalPlanDistance"       value="$(arg maxLocalPlanDistance)" />
    <param name="samplingTipMargin"       value="$(arg samplingTipMargin)" />
    <param name="samplingOutMargin"       value="$(arg samplingOutMargin)" />
    <param name="samplingSpeedFactor"       value="$(arg samplingSpeedFactor)" />
    <param name="pathDensity"           value="$(arg pathDensity)" />
    <param name="rollOutDensity"         value="$(arg rollOutDensity)" />
    <param name="rollOutsNumber"         value="$(arg rollOutsNumber)" />
    <param name="horizonDistance"         value="$(arg horizonDistance)" />
    
    <param name="minFollowingDistance"       value="$(arg minFollowingDistance)" />
    <param name="maxFollowingDistance"       value="$(arg maxFollowingDistance)" />
    <param name="minDistanceToAvoid"       value="$(arg minDistanceToAvoid)" />
    <param name="maxDistanceToAvoid"       value="$(arg maxDistanceToAvoid)" />
    <param name="speedProfileFactor"      value="$(arg speedProfileFactor)" />
    
    <param name="enableSwerving"         value="$(arg enableSwerving)" />
    <param name="enableFollowing"         value="$(arg enableFollowing)" />
    <param name="enableHeadingSmoothing"     value="$(arg enableHeadingSmoothing)" />
    <param name="enableTrafficLightBehavior"   value="$(arg enableTrafficLightBehavior)" />
    <param name="enableLaneChange"         value="$(arg enableLaneChange)" />
    
    <param name="width"             value="$(arg width)" />
    <param name="length"             value="$(arg length)" />
    <param name="wheelBaseLength"         value="$(arg wheelBaseLength)" />
    <param name="turningRadius"         value="$(arg turningRadius)" />
    <param name="maxSteerAngle"         value="$(arg maxSteerAngle)" />
    
    <param name="enableObjectTracking"       value="$(arg enableObjectTracking)" />
    <param name="enableOutsideControl"       value="$(arg enableOutsideControl)" />
    
    <param name="enableOdometryStatus"       value="$(arg enableOdometryStatus)" />
    <param name="mapSource"           value="$(arg mapSource)" />
    <param name="mapFileName"           value="$(arg mapFileName)" />
    
  </node>

</launch>
